tests/decrosstalk/test_pipeline.py

```diff
@@ -50,6 +68,122 @@ def test_run_decrosstalk(tmpdir):
             expected_files.append(roi_name)
             expected_files.append(np_name)
 
+            # Given that it is possible for the code
+            # to gracefully exit saving no ROIs,
+            # verify that all of these traces are not
+            # full of NaNs. Verify that any traces that
+            # are full of NaNs have been flagged as
+            # invalid for some reason
+            for fname in (roi_name, np_name):
+                with h5py.File(fname, 'r') as in_file:
+                    roi_names = in_file['roi_names'][()]
+                    data = in_file['data'][()]
+                    assert not np.isnan(data).all()
+                    assert len(roi_names) > 0
+                    assert data.shape[0] == len(roi_names)
+                    for ii, roi_id in enumerate(roi_names):
+                        if np.isnan(data[ii, :]).any():
+                            assert np.isnan(data[ii, :]).all()
+                            assert roi_id in exp_id_to_invalid[exp_id]
+                        else:
+                            assert roi_id not in exp_id_to_invalid[exp_id]
+
+    output_dir = session['qc_output_dir']
+    for fname in expected_files:
+        full_name = os.path.join(output_dir, fname)
+        msg = 'could not find %s' % full_name
+        assert os.path.isfile(full_name), msg
+
+
+def test_run_decrosstalk_with_empty_rois(tmpdir):
```

Comment:
these 2 long tests with different inputs/expectations would be a good use case for injecting arguments via a pytest fixture.
but, for another time, if you need this to go through for release.

Reply:
Good point. I need to get more familiar with pytest's bells and whistles.

As you say, though: after things calm down.